sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -112,8 +113,12 @@
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<ConfigurationSetting> addSetting(ConfigurationSetting setting) {
-        return withContext(context -> addSetting(setting, context))
-            .flatMap(response -> Mono.justOrEmpty(response.getValue()));
+        try {
```

Comment:
Where is the the RuntimeException being thrown from? I'm surprised that the `withContext(…)` or underlying RestProxy isn't bubbling this to downstream.

Reply:
The exception is thrown in one of the methods in this class `validateSetting(setting)`. Besides, the purpose of this change is to catch all runtime exceptions before the user code - sdk api handshake and return them in the error channel.